### PR TITLE
Update ApparentPowerConstraintUtil

### PR DESCRIPTION
**Pull request by Sn0w3y**
The `getConstraintThroughPoints` method performed division by `(p2.x - p1.x)` without checking if the denominator equals zero. This could occur when two consecutive points on the apparent power circle have the same x-coordinate (active power).

**Impact:** 
- Division by zero causing `Infinity` or `NaN` values
- Incorrect power distribution in energy management
- Potential system crashes during power optimization

### Fix Implemented

1. **Added division-by-zero check** using epsilon comparison (`1e-10`) for floating-point precision
2. **Special handling for vertical lines** - when points have the same x-coordinate, create a constraint that directly constrains active power
3. **Comprehensive test suite** with 6 test cases covering:
   - Normal operation with positive apparent power
   - Edge cases (zero, negative, small, large values)
   - Vertical line scenarios
   - Validation that all constraint values are finite (no NaN/Infinity)

The fix ensures the power optimization system can handle all geometric configurations of the apparent power circle without mathematical errors.

### CODE REVIEW

File: io.openems.edge.ess.core/src/io/openems/edge/ess/core/power/data/ApparentPowerConstraintUtil.java [hunk 1]
```diff
@@ -91,8 +91,20 @@ private static Constraint getConstraintThroughPoints(Coefficients coefficients,
 		 *  y = ((y2-y1)/(x2-x1)) * x + ((x2*y1-x1*y2)/(x2-x1))
 		 * </pre>
 		 */
-		var constraintValue = -1 * (p1.y * p2.x - p2.y * p1.x) / (p2.x - p1.x);
-		var coefficient1 = (p2.y - p1.y) / (p2.x - p1.x);
+		var deltaX = p2.x - p1.x;
+
+		// Check for division by zero - if points have same x-coordinate, create
+		// vertical constraint
+		if (Math.abs(deltaX) < 1e-10) {
```
Comment (da-Kai) [inline io.openems.edge.ess.core/src/io/openems/edge/ess/core/power/data/ApparentPowerConstraintUtil.java]: I feel like this could be extracted as a function in [DoubleUtils](io.openems.common/src/io/openems/common/utils/DoubleUtils.java) and `1e-10` as a private constant within DoubleUtils.

**Review by sfeilmeier — APPROVED** [review]
